File: jimmer-example/spring-boot-with-jimmer-multiple-datasource/src/main/java/org/jimmer/jimmerdatasouce/domain/Dept.java
package org.jimmer.jimmerdatasouce.domain;

import org.babyfish.jimmer.sql.*;

/**
 * @author jhlz
 * @version x.x.x
 */
@Entity
@Table(name = "tb_dept")
public interface Dept {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    long deptId();

    String deptName();

    Long leaderId();

    Long roleId();
}

File: jimmer-example/quarkus-with-jimmer-kt/src/main/kotlin/org/jimmer/domain/Dept.kt
package org.jimmer.domain

import org.babyfish.jimmer.sql.*

/**
 * @author jhlz
 * @since 2024/4/17 13:48
 * @version x.x.x
 */
@Entity
@Table(name = "tb_dept")
interface Dept {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val deptId: Long
    val deptName: String
    val leaderId: Int?

    val roleId: Int?
}